fbw-a380x/src/systems/instruments/src/ND/OansControlPanel.tsx

```diff
@@ -823,16 +830,24 @@ export class OansControlPanel extends DisplayComponent<OansProps> {
                   <div class="oans-cp-status-2">
                     <Button
                       label="SWAP"
-                      disabled={Subject.create(true)}
-                      onClick={() => console.log('SWAP')}
+                      disabled={this.navigraphAvailable}
+                      onClick={() => {
+                        this.amdbClient
+                          .searchForAirports('')
+                          .then((airports) => {
+                            this.store.airports.set(airports);
+                            this.navigraphAvailable.set(true);
+                          })
+                          .catch(() => this.navigraphAvailable.set(false));
+                      }}
```

Comment:
It would be nicer to move non-trivial logic into a proper method rather than an inline function.